src/OpenTelemetry.Exporter.OpenTelemetryProtocol.PersistentStorage/OtlpTraceExporterPersistentStorageExtensions.cs

```diff
@@ -0,0 +1,95 @@
+    /// <param name="name">Name which is used when retrieving options.</param>
+    /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <param name="persistentStorageFactory">Factory function to create a <see cref="PersistentBlobProvider"/>.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
+        this TracerProviderBuilder builder,
+        string? name,
+        Action<OtlpExporterOptions> configure,
+        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
+    {
+        Guard.ThrowIfNull(persistentStorageFactory);
+
+        Action<OtlpExporterOptions, IServiceProvider>? inlineConfigurationAction;
+        if (name is not null)
+        {
+            // Note: If we are using named options we can safely use Options API
+            // to act on the instance that will be created. See:
+            // https://github.com/open-telemetry/opentelemetry-dotnet/issues/4043
+            builder.ConfigureServices(services =>
+            {
+                services
+                      .AddOptions<OtlpExporterOptions>(name)
+                      .Configure<IServiceProvider>(ConfigureOptionsAction);
+            });
+
+            inlineConfigurationAction = null;
+        }
+        else
+        {
+            // Note: If we are NOT using named options we need to execute inline
+            // to prevent potentially impacting options for other signals.
+            inlineConfigurationAction = ConfigureOptionsAction;
+        }
+
+        return builder.AddOtlpExporter(
+            name,
+            configure,
+            inlineConfigurationAction);
+
+        void ConfigureOptionsAction(OtlpExporterOptions options, IServiceProvider serviceProvider)
+        {
+            options.PersistentBlobProvider = persistentStorageFactory(serviceProvider);
+        }
+    }
+
+    /// <summary>
+    /// Adds OpenTelemetry Protocol (OTLP) exporter to the TracerProvider
+    /// with access to persistent storage.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> builder to use.</param>
+    /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+    /// <param name="persistentStorageFactory">Factory function to create a <see cref="PersistentBlobProvider"/>.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
+        this TracerProviderBuilder builder,
+        Action<OtlpExporterOptions> configure,
+        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
+    {
+        return builder.AddOtlpExporterWithPersistentStorage(name: null, configure, persistentStorageFactory);
+    }
```

Comment:
If you do make the configure delegate nullable (optional) on line 42 consider doing this...

```csharp
    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
        this TracerProviderBuilder builder,
        Action<OtlpExporterOptions> configure,
        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
    {
        Guard.ThrowIfNull(configure);
        return builder.AddOtlpExporterWithPersistentStorage(name: null, configure, persistentStorageFactory);
    }

    public static TracerProviderBuilder AddOtlpExporterWithPersistentStorage(
        this TracerProviderBuilder builder,
        Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
    {
        return builder.AddOtlpExporterWithPersistentStorage(name: null, configure: null, persistentStorageFactory);
    }
```

That just makes the overloads friendly to having the configure delegate be optional.